gprofiler/main.py

```diff
@@ -382,6 +373,74 @@ def run_continuous(self) -> None:
                     break
 
 
+def _submit_profile_logged(
+    client: APIClient,
+    start_time: datetime.datetime,
+    end_time: datetime.datetime,
+    profile: str,
+    total_samples: int,
+    profile_api_version: Optional[str],
+    spawn_time: float,
+    metrics: "Metrics",
+    gpid: str,
+) -> Dict:
+    try:
+        response_dict = client.submit_profile(
+            start_time,
+            end_time,
+            profile,
+            total_samples,
+            profile_api_version,
+            spawn_time,
+            metrics,
+            gpid,
+        )
+        gpid = response_dict.get("gpid", "")
+    except Timeout:
+        logger.error("Upload of profile to server timed out.")
+    except APIError as e:
+        logger.error(f"Error occurred sending profile to server: {e}")
+    except RequestException:
+        logger.exception("Error occurred sending profile to server")
+    else:
+        logger.info("Successfully uploaded profiling data to the server")
+
+
+def send_collapsed_file_only(args, client):
+    spawn_time = time.time()
+    gpid = ""
+    metrics = NoopSystemMetricsMonitor().get_metrics()
+    if args.collect_metadata:
+        _static_metadata = get_static_metadata(spawn_time=spawn_time, run_args=args.__dict__)
+    metadata = (
+        get_current_metadata(cast(Metadata, _static_metadata))
+        if args.collect_metadata
+        else {"hostname": get_hostname()}
+    )
+    # TODO:container names, application metadata
+    local_start_time, local_end_time, merged_result, total_samples = concatenate_from_external_file(
+        args.file_path,
+        metadata,
+    )
+
+    if local_start_time is None and local_end_time is None:
+        assert (
+            local_start_time is None and local_end_time is None
```

Comment:
Assert is the same condition as the `if`, so it will never be triggered. I suppose the `if` is meant to be `local_start_time is None or local_end_time is None`? (notice the `and` -> `or` diff)